### PR TITLE
[GH-#116] When being added by a new peer, try to add it as well

### LINKED ISSUE #116: when being added by a new peer, try to add it as well
- [x] when a peer we don't know yet calls our `/info` find his uri (phenix should have some kind of method) and try to add him as well, if we don't know him and if the genesis_block matches
- [x] maybe check if it is a aeclient type of peer through `server` response header

### PR DESCRIPTION
**Pull request by d-velev**
closes #116

### CODE REVIEW

File: apps/aehttpserver/web/controllers/info_controller.ex [hunk 1]
```diff
@@ -18,11 +20,27 @@ defmodule Aehttpserver.InfoController do
...
+    if(!(peer == host || host == "localhost:4000")) do
+      case(Map.has_key?(Peers.all_peers, peer)) do
+        true ->
+          Logger.info("Peer already in our list")
+        false ->
+          Peers.schedule_add_peer(peer)
+      end
+    end
+
+    conn = Plug.Conn.put_resp_header(conn, "server", "aehttpserver")
```
Comment (thepiwo) [inline apps/aehttpserver/web/controllers/info_controller.ex]: can you put this to a general place, for all endpoints to be active, or does this have to exactly here?

File: apps/aecore/lib/aecore/peers/worker.ex [hunk 2]
```diff
@@ -189,10 +203,13 @@ defmodule Aecore.Peers.Worker do
...
         case own_nonce == info.peer_nonce do
           false ->
-            if(info.genesis_block_hash == genesis_block_header_hash()) do
-              {:ok, info}
-            else
-              {:error, "Genesis header hash not valid"}
+            cond do
+              info.genesis_block_hash != genesis_block_header_hash() ->
+                {:error, "Genesis header hash not valid"}
+              !Map.has_key?(info, :server) || info.server != "aehttpserver"->
```
Comment (cytadela8) [inline apps/aecore/lib/aecore/peers/worker.ex]: Why actually we block peers that are not "aehttpserver"? If someone develops another AE client it's ok. As for validation if we are adding an AE node: we already check genesis_block_hash. If someone returns valid one they probably are an AE node.

File: apps/aehttpclient/lib/client.ex [hunk 3]
```diff
@@ -44,14 +44,17 @@ defmodule Aehttpclient.Client do
...
+      {:ok, %{body: body, headers: headers, status_code: 200}} ->
         case(identifier) do
           :block ->
             response = Poison.decode!(body, as: %Block{}, keys: :atoms!)
             {:ok, response}
           :info ->
             response = Poison.decode!(body, keys: :atoms!)
-            {:ok, response}
+            {_, server} = Enum.find(headers, fn(header) ->
+              header == {"server", "aehttpserver"} end)
```
Comment (cytadela8) [inline apps/aehttpclient/lib/client.ex]: We force `server` header to be equal "aehttpserver" also here. I wonder: what happens if it's not? Do we get an exception that then goes back to `Aecore.Peers.Worker` and causes it to restart?

File: apps/aecore/lib/aecore/peers/worker.ex (hunk 2 above)
Comment (thepiwo) [reply apps/aecore/lib/aecore/peers/worker.ex]: for now it is an easy check to filter other clients (e.g. a wallet) to try to connect

File: apps/aehttpclient/lib/client.ex (hunk 3 above)
Comment (thepiwo) [reply apps/aehttpclient/lib/client.ex]: @d-velev what do you think?

**Pull request merged**